src/reducers/linter.tsx

```diff
@@ -85,3 +105,108 @@ export const getMessageMap = (result: ExternalLinterResult) => {
 
   return msgMap;
 };
+
+export type LinterState = {
+  forVersionId: undefined | number;
+  isLoading: boolean;
+  messageMap: undefined | LinterMessageMap;
+};
+
+export const initialState: LinterState = {
+  forVersionId: undefined,
+  isLoading: false,
+  messageMap: undefined,
+};
+
+export const actions = {
+  abortFetchLinterResult: createAction(
+    'ABORT_FETCH_LINTER_RESULT',
+    (resolve) => {
+      return (payload: { versionId: number }) => resolve(payload);
+    },
+  ),
+  beginFetchLinterResult: createAction(
+    'BEGIN_FETCH_LINTER_RESULT',
+    (resolve) => {
+      return (payload: { versionId: number }) => resolve(payload);
+    },
+  ),
+  loadLinterResult: createAction('LOAD_LINTER_RESULT', (resolve) => {
+    return (payload: { versionId: number; result: ExternalLinterResult }) =>
+      resolve(payload);
+  }),
+};
+
+export const fetchLinterMessages = ({
+  _log = log,
+  url,
+  versionId,
+}: {
+  _log?: typeof log;
+  url: string;
+  versionId: number;
+}): ThunkActionCreator => {
+  return async (dispatch) => {
+    dispatch(actions.beginFetchLinterResult({ versionId }));
+
+    // This is a special URL and returns a non-standard JSON response.
+    const response = await fetch(url);
+
+    try {
+      if (!response.ok) {
+        throw new Error(`Got status ${response.status} for URL ${url}`);
+      }
+      const data = await response.json();
+      dispatch(actions.loadLinterResult({ versionId, result: data }));
+    } catch (error) {
+      _log.error(`TODO: handle this error: ${error}`);
+      dispatch(actions.abortFetchLinterResult({ versionId }));
+    }
+  };
+};
+
+export const selectMessageMap = (
+  linterState: LinterState,
+  versionId: number,
+): undefined | LinterMessageMap => {
+  if (linterState.forVersionId !== versionId) {
+    return undefined;
+  }
+  return linterState.messageMap;
+};
+
+const reducer: Reducer<LinterState, ActionType<typeof actions>> = (
+  state = initialState,
+  action,
+): LinterState => {
+  switch (action.type) {
+    case getType(actions.beginFetchLinterResult):
+      return {
+        ...state,
+        forVersionId: action.payload.versionId,
+        messageMap: undefined,
+        isLoading: true,
+      };
+    case getType(actions.abortFetchLinterResult):
+      return {
+        ...state,
+        forVersionId: action.payload.versionId,
+        // TODO: when we have proper error handling, this can
```

Comment:
Is there an issue for that? (or a comment in an existing issue)

Reply:
I can't really think of a helpful issue to file because it all depends on how we implement an error handler. I only added the comment here because it's kind of wrong to load an empty map. It means that the component couldn't re-try an error. But, anyway, retrying an error is already complicated since it wouldn't be an error directly caused by a user action.